File: test/config/app.config.ts
export interface AppConfig {
    deviceId: string;
    appPackage: string;
    appActivity: string;
    appiumServer: {
        hostname: string;
        port: number;
        path: string;
    };
    timeouts: {
        implicit: number;
        explicit: number;
        pageLoad: number;
    };
    capabilities: {
        platformName: string;
        automationName: string;
        noReset: boolean;
    };
}

export const maccabiConfig: AppConfig = {
    deviceId: 'b7a325f6',
    appPackage: 'com.ideomobile.maccabipregnancy',
    appActivity: '.ui.splash.view.PASplashActivity',
    appiumServer: {
        hostname: 'localhost',
        port: 4723,
        path: '/'
    },
    timeouts: {
        implicit: 10000,
        explicit: 30000,
        pageLoad: 60000
    },
    capabilities: {
        platformName: 'Android',
        automationName: 'UiAutomator2',
        noReset: true
    }
};

File: PregnancyApp/Config/AppConfig.cs
namespace PregnancyApp.Config
{
    public class AppConfig
    {
        // Appium Server
        public const string AppiumServerUrl = "http://localhost:4723";

        // App Details
        public const string AppPackage = "com.ideomobile.maccabipregnancy";
        public const string AppActivity = ".ui.splash.view.PASplashActivity";

        // Device Settings
        public const string PlatformName = "Android";
        public const string AutomationName = "UiAutomator2";
        public const string DeviceName = "Android Device";

        // Timeouts (in seconds)
        public const int ImplicitWaitTimeout = 10;
        public const int CommandTimeout = 120;
    }
}
